src/sentry/godot_error_types.h

```diff
@@ -0,0 +1,39 @@
+#ifndef LOGGER_ERROR_TYPES_H
+#define LOGGER_ERROR_TYPES_H
+
+#include "godot_cpp/core/defs.hpp"
+#include "level.h"
+
+#include <godot_cpp/variant/string.hpp>
+
+namespace sentry {
+
+// Godot Engine error types that we can detect.
+// As defined in: https://github.com/godotengine/godot/blob/04692d83cb8f61002f18ea1d954df8c558ee84f7/core/io/logger.h#L51-L56
+enum class GodotErrorType {
+	ERROR_TYPE_ERROR,
+	ERROR_TYPE_WARNING,
+	ERROR_TYPE_SCRIPT,
+	ERROR_TYPE_SHADER,
+};
+
+// Enum used with bitwise operations to represent the set of Godot error types that the Sentry logger should capture.
+enum class GodotErrorMask {
+	NONE = 0,
+	ERROR = 1 << int(GodotErrorType::ERROR_TYPE_ERROR),
+	WARNING = 1 << int(GodotErrorType::ERROR_TYPE_WARNING),
+	SCRIPT = 1 << int(GodotErrorType::ERROR_TYPE_SCRIPT),
+	SHADER = 1 << int(GodotErrorType::ERROR_TYPE_SHADER),
+	ALL = ERROR | WARNING | SCRIPT | SHADER,
+	ALL_EXCEPT_WARNING = ERROR | SCRIPT | SHADER,
```

Comment:
is `SCRIPT` very spammy? We'd want the default behavior to only capture events for something you really want to act on.

Reply:
Any of those can be spammy. That largely depends on the project.

Reply:
In any regard you might want to make it less spammy as you can. So keeping just `ERROR` perhaps is the right call, instead of adding also the other two. 

I'm not familiar with this enough but if there are too many entries, it'll be noisy and incur overhead

Reply:
To clarify things, `SCRIPT` are errors that occur in a GDScript code, and `ERROR` are for errors that occur in the native code (which could still originate in a GDScript call). 
Spammy errors is largely a matter of fixing these issues in the project. It can be spammy if errors are simply ignored by the developer. If it's a serious concern, perhaps keeping it disabled by default is the better choice, making it intentional if the dev wants it in their project.

Reply:
We've got the same issue in Unity with errors originating from within the `Update` loop. But we've got a "not so great" debouncer in place for people to opt-in if they so wish.
https://github.com/getsentry/sentry-godot/issues/22

Reply:
I like the debouncer idea. We can throttle the misbehaving parts and avoid silencing the rest of the user's project.